### ISSUE TITLE
amp-social-share: No indication of focus

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#29561]
## WCAG Level
Level AA

## Priority
High

## Type
Navigation

## Pages/screens/Components affected
* [amp-social-share](https://amp.dev/documentation/components/amp-social-share/)

## Description
Keyboard users rely on the browser's focus outline to keep track of where they are on web pages. The currently selected form input, button, or hyperlink is the one said to have the focus.

Most browsers use a faint dotted outline to indicate the focus to the user, though some browsers like Chrome or Safari use an enhanced focus outline, consisting of a coloured border around the selected link. It is also possible for developers to provide custom focus indications within the site's cascading style sheets (CSS).

Currently, the `<amp-social-share class="rounded">` variant of the buttons in the [playground example](https://playground.amp.dev/?url=https%3A%2F%2Fpreview.amp.dev%2Fdocumentation%2Fexamples%2Fcomponents%2Famp-social-share&format=websites&_gl=1*1nre47m*_ga*YW1wLWdDQmZ0cDgwZl96MWxoRWR1NEFlaXc.) lack any visible focus indication in Firefox. While the regular/square buttons show Firefox's default dotted outline (which is arguably still hard to discern, but nominally passes the "visible" requirement of WCAG 2.1 SC 2.4.7 Focus Visible), the rounded buttons lack any such indication.

![The social share buttons in Firefox, with a square version focused to show the dotted outline](https://user-images.githubusercontent.com/54933393/88787492-d99fa880-d18b-11ea-9d39-bc33d297cb20.png)

It's unclear if the `rounded` class is just an example of a custom style that authors *can* define themselves, or if this is a core part/ships by default with `<amp-social-share>` (if it is, it's not covered in the documentation). Either way, this is a currently a failure.

## User impact
Without a visible focus indicator it can be difficult or impossible for keyboard users that aren't using additional assistive technology such as a screen reader to know which control (whether it be a link, button, or something else) currently has focus and therefore what action will take place if they press the `Enter`, `Space`, or other keys.

## Required solution
Make sure that all focusable elements have a clear focus style. In this case, make sure that an explicit focus indication has been defined for the rounded social share buttons. Optionally, consider setting a consistent focus style for all social share buttons, including the regular rectangular variants. For instance, you can explicitly set the outline with

```
amp-social-share.rounded:focus {
    outline: 2px var(--color-primary)solid;
    outline-offset: 2px;
}
```

![The social share buttons in Firefox, with the proposed CSS applied - rounded buttons now showing a solid blue focus outline](https://user-images.githubusercontent.com/54933393/88787511-e02e2000-d18b-11ea-94ed-ca8da948447b.png)

To only make these styles apply for keyboard users, it is possible to use [`:focus-visible`](https://drafts.csswg.org/selectors-4/#the-focus-visible-pseudo) instead of `:focus`. However, support for this is still not pervasive across browsers, and will require the use of a workaround, such as the [WICG focus-visible polyfill](https://github.com/WICG/focus-visible).

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied:
1. Use a browser such as Firefox which has does not have a highly visible default focus indication.
2. Navigate to the rounded share buttons on the page using the keyboard (`Tab`/`Shift`+`Tab`).
3. Check that the focus indicator is visible.

## Related standards
* [WCAG 2.1 Success Criterion 2.4.7 Focus Visible (Level AA)](https://www.w3.org/TR/WCAG21/#focus-visible)

## More information
* [W3C - Failure of Success Criterion 2.4.7 due to styling element outlines and borders in a way that removes or renders non-visible the visual focus indicator](https://www.w3.org/WAI/WCAG21/Techniques/failures/F78.html)

## Test data
Test date: July 2020

**Comment by caroqliu** [comment]
Fixed by #30775
